File: lab5/C++/src/Weapon.cpp
#include "Weapon.hpp"
#include "SupplyRequestDetail.hpp"


Weapon::Weapon(const Weapon& other) :
    Equipment(rand(), other.GetName(), other.GetCount()), m_caliber(other.m_caliber) {
}

Weapon::Weapon(int id, const std::string& name, Caliber caliber, int count)
    : Equipment(id, name, count), m_caliber(caliber) { }

bool Weapon::IsMatches(const SupplyRequestDetail& detail) const {
    return detail.GetSupplyType() == SupplyType::Weapon && detail.GetCaliber() == m_caliber;
}

Caliber Weapon::GetCaliber() const {
    return m_caliber;
}

File: lab6/Java/src/warehouse/equipments/Weapon.java
package warehouse.equipments;

import enums.Caliber;
import enums.SupplyType;
import request.SupplyRequestDetail;

public class Weapon extends Equipment {
    Caliber caliber;
    public Weapon(int id, String name, Caliber caliber, int count)
    {
        super(id, name, count);
        this.caliber = caliber;
    }

    @Override
    public Boolean isMatches(SupplyRequestDetail detail) {
        return detail.getSupplyType() == SupplyType.WEAPON &&
               detail.getCaliber() == caliber;
    }

    public Caliber getCaliber() {
        return caliber;
    }
}
